app/services/kit_create_service.rb

```diff
@@ -0,0 +1,102 @@
+class KitCreateService
+  include ServiceObjectErrorsMixin
+
+  attr_reader :kit
+
+  def initialize(organization_id:, kit_params:)
+    @organization_id = organization_id
+    @kit_params = kit_params
+  end
+
+  def call
+    return self unless valid?
+
+    organization.transaction do
+      # Create the Kit record
+      @kit = Kit.new(kit_params_with_organization)
+      @kit.save!
+
+      # Create a BaseItem that houses each
+      # kit item created.
+      kit_base_item = fetch_or_create_kit_base_item
+
+      # Create the Item.
+      item_creation = ItemCreateService.new(
+        organization_id: organization.id,
+        item_params: {
+          name: kit.name,
+          partner_key: kit_base_item.partner_key,
+          kit_id: kit.id
+        }
+      )
+
+      item_creation.call
+    rescue StandardError => e
+      errors.add(:base, e.message)
+      raise ActiveRecord::Rollback
+    end
+
+    self
+  end
+
+  private
+
+  attr_reader :organization_id, :kit_params
+
+  def organization
+    @organization ||= Organization.find_by(id: organization_id)
+  end
+
+  def kit_params_with_organization
+    kit_params.merge({
+                       organization_id: organization.id
+                     })
+  end
+
+  def fetch_or_create_kit_base_item
+    BaseItem.find_or_create_by!({
+                                  name: 'Kit',
+                                  category: 'kit',
+                                  partner_key: 'kit'
+                                })
```

Comment:
Since this is in a `BaseItem` which is global for all and immutable except by SuperAdmin users, this should probably be added to the base item seeds, added manually in the console / in prod. You could likely define it as a class constant, and reference that in the seed / migration.

Once this runs a single time, it will never need to be created again

Reply:
@armahillo good point i"ll add that to the seeds.rb